src/python/pants/backend/native/subsystems/native_build_step.py

```diff
@@ -36,15 +36,15 @@ def register_options(cls, register):
                   'for targets of this language.')
 
     register_enum_option(
-      register, ToolchainVariant, '--toolchain-variant', advanced=True,
+      register, ToolchainVariant, '--toolchain-variant', advanced=True, default='gnu',
```

Comment:
Why not `default=ToolchainVariant.gnu`? I prefer the explicit link, rather than the reader needing to trace through "The default is a string, but the type is an enum, how do those relate to each other"

Reply:
It's also great because the explicit attribute access also validates that it's a valid value for the enum (which is already done in `choices`, but more clear this way). Will do.

Reply:
The main reason why not right now actually is because if you use enum objects for the option values, you run into #7233 (if the user provides an invalid choice, the error message doesn't include the option name or scope). I will see how much extra code it would take to fix #7233 here and perhaps make a separate PR.

Reply:
(I will make a separate PR)